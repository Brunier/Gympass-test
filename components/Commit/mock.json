{
  "sha": "942a765b53662936679b8119ed4533d972298d2a",
  "commit": {
    "author": {
      "name": "Pedro Bullo Junior",
      "email": "pedro.bj@hotmail.com",
      "date": "2019-01-21T22:59:19Z"
    },
    "committer": {
      "name": "GitHub",
      "email": "noreply@github.com",
      "date": "2019-01-21T22:59:19Z"
    },
    "message": "Workaround empty child",
    "tree": {
      "sha": "81c34b008db2a4fda4df20404a498e0127e4a817",
      "url": "https://api.github.com/repos/Gympass/react-native-svg-uri/git/trees/81c34b008db2a4fda4df20404a498e0127e4a817"
    },
    "url": "https://api.github.com/repos/Gympass/react-native-svg-uri/git/commits/942a765b53662936679b8119ed4533d972298d2a",
    "comment_count": 0,
    "verification": {
      "verified": true,
      "reason": "valid"
    }
  },
  "url": "https://github.com/Gympass/react-native-svg-uri/commit/942a765b53662936679b8119ed4533d972298d2a"
}
