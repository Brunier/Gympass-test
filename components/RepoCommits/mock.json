{
  "data" : [
    {
      "sha": "942a765b53662936679b8119ed4533d972298d2a",
      "commit": {
        "author": {
          "name": "Pedro Bullo Junior",
          "email": "pedro.bj@hotmail.com",
          "date": "2019-01-21T22:59:19Z"
        },
        "committer": {
          "name": "GitHub",
          "email": "noreply@github.com",
          "date": "2019-01-21T22:59:19Z"
        },
        "message": "Workaround empty child",
        "tree": {
          "sha": "81c34b008db2a4fda4df20404a498e0127e4a817",
          "url": "https://api.github.com/repos/Gympass/react-native-svg-uri/git/trees/81c34b008db2a4fda4df20404a498e0127e4a817"
        },
        "url": "https://api.github.com/repos/Gympass/react-native-svg-uri/git/commits/942a765b53662936679b8119ed4533d972298d2a",
        "comment_count": 0,
        "verification": {
          "verified": true,
          "reason": "valid"
        }
      },
      "url": "https://github.com/Gympass/react-native-svg-uri/commit/942a765b53662936679b8119ed4533d972298d2a"
    },
    {
      "sha": "c6841a6e7d1a6ee5cf5b515d4faea6002e166f3c",
      "commit": {
        "author": {
          "name": "matc4",
          "email": "matias.s.cortes@gmail.com",
          "date": "2018-09-13T12:34:04Z"
        },
        "committer": {
          "name": "GitHub",
          "email": "noreply@github.com",
          "date": "2018-09-13T12:34:04Z"
        },
        "message": "Merge pull request #100 from nesc58/masteFix: Text elements / transform + clipPath attribute added",
        "tree": {
          "sha": "97dae56ead52ab59e12b98848f4e38d89265040f",
          "url": "https://api.github.com/repos/Gympass/react-native-svg-uri/git/trees/97dae56ead52ab59e12b98848f4e38d89265040f"
        },
        "url": "https://api.github.com/repos/Gympass/react-native-svg-uri/git/commits/c6841a6e7d1a6ee5cf5b515d4faea6002e166f3c",
        "comment_count": 0,
        "verification": {
          "verified": true,
          "reason": "valid"
        }
      },
      "url": "https://github.com/Gympass/react-native-svg-uri/commit/c6841a6e7d1a6ee5cf5b515d4faea6002e166f3c"
    }
  ]
}
