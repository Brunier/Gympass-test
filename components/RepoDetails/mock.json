{
  "name": "interview-test",
  "description": "Interview test project",
  "url": "https://api.github.com/repos/Gympass/interview-test",
  "commits_url": "https://api.github.com/repos/Gympass/interview-test/commits",
  "created_at": "2018-05-22T19:57:37Z",
  "updated_at": "2019-01-24T02:51:23Z",
  "language": "Javascript",
  "stars": 3,
  "watchers": 3,
  "forks": 4
}
