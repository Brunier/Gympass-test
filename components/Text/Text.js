import styled from 'styled-components';
import React from 'react';



let Text = styled.span`
  color: #585656;
  font-size: ${props => props.fontSize ? props.fontSize + "px"  : "20px"};
  font-weight: ${props => props.bold ? "bold" : "none"};
`

export default Text;

