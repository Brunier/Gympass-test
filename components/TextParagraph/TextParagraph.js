import styled from 'styled-components';
import React from 'react';


let TextParagraph = styled.p`
  overflow: hidden;
  position: relative;
  line-height: 16px;
  max-height: ${props => props.maxHeight ? props.maxHeight + "px"  : "96px"}; 
  height: ${props => props.height ? props.height + "px" : "inherit"}; 
  margin-right: -1em;
  padding-right: 1em;
  color: #585656;
`

export default TextParagraph;


/*
 color: #585656;
  font-size: 16px;
    height: 96px;
    overflow: hidden;
    text-overflow: ellipsis;
    max-height: 90px;
 */
