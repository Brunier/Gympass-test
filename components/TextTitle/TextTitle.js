import styled from 'styled-components';
import React from 'react';

let TextTitle = styled.h1`
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
    max-width: ${props => props.maxWidth ? props.maxWidth : "90%"};
    display: ${props => props.display ? props.display : "block"};
    font-size: ${props => props.fontSize ? props.fontSize + "px" : "20px"};
    color: #fff;
    padding: ${props => props.padding ? props.padding : "5px"};
    background: #00757e;
    margin-top: 0 
`

export default TextTitle;

