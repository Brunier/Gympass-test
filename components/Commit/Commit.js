import React from 'react';
import styled from "styled-components";
import Text from "../Text/Text";
import TextTitle from "../TextTitle/TextTitle";
import TextParagraph from "../TextParagraph/TextParagraph";
import moment from "moment/moment";

const Container = styled.div`
  padding: 10px;
  border: 2px solid #00757e;
  height: 200px;
  margin-bottom: 30px;
    
  a {
    text-decoration: none;
    color: inherit;
  }
  
 `;

export class Commit extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return(
            <Container>
                <div>
                    <TextTitle maxWidth={"100%"} fontSize={18}><a href={this.props.commit.url} target="_blank">{this.props.commit.sha}</a></TextTitle>
                    <TextParagraph maxHeight={120} height={120}>
                        {this.props.commit.commit.message}
                    </TextParagraph>
                </div>
                <div>
                    <Text fontSize={16} bold={true}>{this.props.commit.commit.author.name}</Text>
                    <Text fontSize={16} bold={true}> - </Text>
                    <Text fontSize={16} bold={true}>{moment(this.props.commit.commit.author.date).format('llll')}</Text>
                </div>
            </Container>
        )
    }
}

export default Commit;
