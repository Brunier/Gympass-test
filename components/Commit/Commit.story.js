import React from 'react';
import { storiesOf } from '@storybook/react';
import Commit from './Commit';
import mock from './mock'

storiesOf('Commit', module)
    .add('Normal', () => (
        <Commit commit={mock} />
    ))
