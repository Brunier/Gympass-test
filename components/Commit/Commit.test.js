import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer'

// Components
import Commit from './Commit';
import Text from "../Text/Text";
import TextTitle from "../TextTitle/TextTitle";
import TextParagraph from "../TextParagraph/TextParagraph";
import moment from "moment/moment";
import mock from "./mock.json"


function setup(props) {
    const wrapper = shallow(<Commit {...props} />);
    return { wrapper };
}


const commit = mock;


describe('Commit Test Suite', () => {
    it('Titulo do commit', () => {
        const { wrapper } = setup({commit});
        expect(wrapper.find(TextTitle).text()).toBe(commit.sha);
    });
    it('Mensagem do commit', () => {
        const { wrapper } = setup({commit});
        expect(wrapper.find(TextParagraph).text()).toBe(commit.commit.message);
    });
    it('Nome do autor do commit', () => {
        const { wrapper } = setup({commit});
        expect(wrapper.find(Text).at(0).text().trim()).toBe(commit.commit.author.name);
    });
    it('Data do commit', () => {
        const { wrapper } = setup({commit});
        expect(wrapper.find(Text).at(2).text().trim()).toBe(moment(commit.commit.author.date).format('llll'));
    });

    it('Snapshot', () => {
        const tree = renderer.create(<Commit commit={commit}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
});
