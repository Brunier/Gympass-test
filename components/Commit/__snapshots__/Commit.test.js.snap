// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Commit Test Suite Snapshot 1`] = `
<div
  className="sc-bxivhb bqVlmP"
>
  <div>
    <h1
      className="sc-bwzfXH bJRMPj"
      fontSize={18}
    >
      <a
        href="https://github.com/Gympass/react-native-svg-uri/commit/942a765b53662936679b8119ed4533d972298d2a"
        target="_blank"
      >
        942a765b53662936679b8119ed4533d972298d2a
      </a>
    </h1>
    <p
      className="sc-htpNat jewUiJ"
      height={120}
    >
      Workaround empty child
    </p>
  </div>
  <div>
    <span
      className="sc-bdVaJa iDsqdx"
      fontSize={16}
    >
      Pedro Bullo Junior
    </span>
    <span
      className="sc-bdVaJa iDsqdx"
      fontSize={16}
    >
       - 
    </span>
    <span
      className="sc-bdVaJa iDsqdx"
      fontSize={16}
    >
      Mon, Jan 21, 2019 8:59 PM
    </span>
  </div>
</div>
`;
